### ISSUE TITLE
Feature Request: Option to Alert GMs when Token Lean is used

### ISSUE THREAD

**Issue by Genkitty** [WesBelmont/token-lean#15]
I am all in favor of allowing my players to use Token Lean, but I would really like to know when it is happening so I have a chance to check to see if anyone notices the leaning character.  Having the token twitch isn't enough for my needs, as I'm not always paying full attention to the map (GM tasks).

**Comment by WesBelmont** [comment]
What would be your preffered method of notification? I could add a chat hook or popup pretty easily, but I feel like that would crowd the interface quickly. Would a sound be sufficient?

**Comment by gsterling** [comment]
I have some ideas (I don't know if these are possible since it looks like leaning happens on the client):
- Highlight the area the token sees by leaning, like Perfect Vision does with it's 'Delimiters'
- Create and move a transparent version of the token only visible for the GM on top of the old one where the "head" is. So we see them going around the corner
- Add a request for leaning, that the GM has to approve
- Share vision from leaning with the GM. This isn't as noticable, but you at least can get an idea what you players saw, if they can cooperate well with the GM
- Don't allow players to lean themselves but instead have the GM do it for them and share the new vision with them.
- Drop a marker on the map, where a token leaned, so you can react on it afterwards.

Personally I prefer the approach of them to request the leaning and sharing the vision with me after allowing it. That way I can have them roll for stealth before they can see anything. This changes the current smoother behaviour quite a bit, but sometimes I just don't want them to look around a corner without me noticing.

If you'd want to do it really fancy have a toggle for free leaning and approved leaning on a per map basis or even have different zones with the draw tool where you can activate the approvals, like Perfect Vision does it with it's lightning settings.

**Comment by WesBelmont** [comment]
The transparent token is something that will likely be coming in version 2.0. Dropped markers where leaning happened is a good idea but I'll need to dig more thouroughly into foundry/PIXI to get it to work nicely. I will be able to add a global toggle pretty easily so I'll do that in the meantime.

A gm inverted fog of war that shows everything the players have seen is a neat idea, but feels beyond the scope of the module. If someone made it, I'd endeavour to maintain compatability.

**Comment by gsterling** [comment]
> A gm inverted fog of war that shows everything the players have seen is a neat idea, but feels beyond the scope of the module. If someone made it, I'd endeavour to maintain compatability.

Yes, I agree. I didn't want it to remove fog, just mark the area that was viewed by the player.

Something like this overlay:
![image](https://user-images.githubusercontent.com/58980630/195874692-f1e733e4-7dfd-434a-87ae-23d150963a3a.png)

I think this would fit in this module, since it won't alter the affect the fog, but that doesn't mean it's easier to implement :laughing: 

> I will be able to add a global toggle pretty easily so I'll do that in the meantime.

A global toggle for what exactly?

**Comment by WesBelmont** [comment]
A toggle for enabling/disabling leaning. Then a gm can choose when to allow leaning.
